src/main/java/duck/Duck.java

```diff
@@ -0,0 +1,51 @@
+package duck;
+
+import java.io.IOException;
+
+import duck.exception.EmptyDetailsException;
+import duck.exception.UnknownCommandException;
+
+/**
+ * The main class for the DUCK task manager application.
+ * It initializes the required components and handles user interaction.
+ */
+public class Duck {
+
+    private static final String FILE_PATH = "./data/duck.txt";
+
+    /**
+     * The main entry point of the DUCK application.
+     *
+     * @param args Command-line arguments (not used).
+     * @throws EmptyDetailsException If a task description is missing.
+     * @throws UnknownCommandException If an unrecognized command is entered.
+     * @throws IOException If an error occurs while reading or writing the storage file.
+     */
+    public static void main(String[] args) throws EmptyDetailsException, UnknownCommandException, IOException {
```

Comment:
main Method Should Not Throw Exceptions Directly. If an exception is thrown, it crashes the program without proper error handling. Other than this LGTM!
```suggestion
    public static void main(String[] args) {
    try {
        new Duck().run();
    } catch (IOException e) {
        System.err.println("Error accessing storage file: " + e.getMessage());
    } catch (Exception e) {
        System.err.println("An unexpected error occurred: " + e.getMessage());
    }
}
```